EIPS/eip-abstract_wallet_transaction_api.md

```diff
@@ -0,0 +1,218 @@
+---
+eip: <to be assigned>
+title: Abstract Wallet Transaction API
+description: JSON RPC methods for dealing with transactions that are more suitable for different kinds of wallets
+author: Moody Salem (@moodysalem)
+discussions-to: <URL>
+status: Draft
+type: Meta
+created: 2022-10-17
+---
+
+## Abstract
+
+Defines new JSON RPC methods for dapps to send messages from a user's wallet, as well as check on the status
+of these messages. These methods are more abstract to allow for differences in how messages are sent on-chain from the
+user's wallet, e.g. smart contract wallets utilizing [EIP-4337](./eip-4337.md) or EOA wallets that
+support bundled transactions via [EIP-3074](./eip-3074.md).
+
+## Motivation
+
+The current APIs to send transactions and check their status, `eth_sendTransaction` and `eth_getTransactionReceipt`,
+are not flexible enough for dapps to provide good notifications and user experience for transaction status.
+For example, these methods do not allow sending multiple transactions, or querying the status of a replaced-by-fee (sped
+up)
+transaction.
```

Comment:
Technically you can query the status of a replaced-by-fee transaction by subscribing to pending and mined transactions and comparing nonces, right? It's not exactly trivial though.

Reply:
That's a ton of work for the dapp to do, essentially you're indexing a firehose just to figure out something the wallet trivially knows because it initiated the change